app/src/main/java/com/github/se/project/model/authentification/AuthenticationViewModel.kt

```diff
@@ -24,120 +24,142 @@ import kotlinx.coroutines.flow.callbackFlow
 
-        // Check if the received credential is a valid Google ID Token
+        // Check if the credential is a valid Google ID Token
         if (credential is CustomCredential &&
             credential.type == GoogleIdTokenCredential.TYPE_GOOGLE_ID_TOKEN_CREDENTIAL) {
           // Extract the Google ID Token credential
           val googleIdTokenCredential = GoogleIdTokenCredential.createFrom(credential.data)
-          // Create an auth credential using the Google ID Token
+
+          // Create an AuthCredential from the Google ID Token
           val authCredential =
               GoogleAuthProvider.getCredential(googleIdTokenCredential.idToken, null)
-          // Sign in with Firebase using the auth credential
-          val authResult =
-              firebaseAuth
-                  .signInWithCredential(authCredential)
-                  .await() // .await() -> allows the coroutine to wait for the result of the
-          // authentication operation before proceeding.
-          // Send the successful result
-          trySend(
-              Result.success(
-                  authResult)) // Is used to send the result of the Firebase sign-in operation to
-          // the Flow's collectors.
+
+          // Sign in with Firebase using the AuthCredential
+          val authResult = firebaseAuth.signInWithCredential(authCredential).await()
+
+          // Emit the result as successful sign-in
+          trySend(Result.success(authResult))
         } else {
-          // Throw an exception if the credential type is invalid
+          // If credential is invalid, log the error
           Log.e("AuthenticationViewModel", "Invalid credential type received.")
           throw RuntimeException("Received an invalid credential type.")
         }
       } catch (e: GetCredentialCancellationException) {
-        // Handle sign-in cancellation
+        // Handle case where sign-in is canceled by the user
         trySend(Result.failure(Exception("Sign-in was canceled.")))
       } catch (e: Exception) {
-        // Handle other exceptions
+        // Log and emit error for any other exceptions
         Log.e("AuthenticationViewModel", "Error during authentication process", e)
         trySend(Result.failure(e))
       }
 
-      // When a collector starts collecting from the callbackFlow, the flow remains open and ready
-      // to emit values until the awaitClose block is reached or the flow is cancelled.
-      // Even though the current block is empty, in other scenarios, you might use the awaitClose
-      // block to unregister listeners or release resources associated with the callback-based API.
+      // Ensure proper closure of the callbackFlow
       awaitClose {}
     }
   }
+
+  /**
+   * Function to handle Firebase sign-out and notify when done.
+   *
+   * @param onSignOutComplete Lambda function that will be triggered when the sign-out process is
+   *   complete.
+   */
+  fun signOut(onSignOutComplete: () -> Unit) {
```

Comment:
The signOut function looks solid and gets the job done! One suggestion: in the catch block, maybe think about handling the error in a way that the caller knows something went wrong, like showing a quick message.